test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +250,99 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases convert release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type == "url" {
+				releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+				if !strings.EqualFold(version.Name, releaseMarker) {
+					ver, err := resolveReleaseMarker(releaseMarker)
+					if err != nil {
+						return fmt.Errorf("failed resolving release path %s with error: %v", version.Name, err)
+					}
+					ver = "v" + ver
+					version.Value = strings.Replace(version.Value, version.Name, ver, 1)
+					version.Name = ver
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func resolveReleaseMarker(releaseMarker string) (string, error) {
```

Comment:
can we have a godoc for this func as well?

Reply:
added some explanation kindly check